v8_c_api/src/v8_c_api.cpp

```diff
@@ -306,6 +307,303 @@ v8_pd_list* v8_PDListCreate(v8::ArrayBuffer::Allocator *alloc) {
+    isolate_ = isolate;
+}
+
+void v8_inspector_channel_wrapper::setOnResponseCallback(const std::function<void(std::string)> &callback) {
+    onResponse_ = callback;
+}
+
+using InspectorOnResponseCallback = std::function<void(std::string)>;
+using InspectorOnWaitFrontendMessageOnPauseCallback = std::function<int(v8_inspector_c_wrapper *)>;
+
+class v8_inspector_client_wrapper final: public v8_inspector::V8InspectorClient {
+public:
+    explicit v8_inspector_client_wrapper(
+        v8::Platform *platform,
+        const v8::Local<v8::Context> &context,
+        const InspectorOnResponseCallback &onResponse = {},
+        const InspectorOnWaitFrontendMessageOnPauseCallback &onWaitFrontendMessageOnPause = {}
+    );
+
+    v8::Local<v8::Context> getContext();
+    v8::Isolate* getIsolate();
+    void setOnResponseCallback(const InspectorOnResponseCallback &callback);
+    void setOnWaitFrontendMessageOnPauseCallback(const InspectorOnWaitFrontendMessageOnPauseCallback &callback);
+
+    void dispatchProtocolMessage(const v8_inspector::StringView &message_view);
+    void runMessageLoopOnPause(const int contextGroupId) override;
+    void quitMessageLoopOnPause() override;
+
+    void schedulePauseOnNextStatement(const v8_inspector::StringView &reason);
+    void waitFrontendMessageOnPause();
+
+private:
+    static const int kContextGroupId = 1;
+
+    v8::Platform* platform_;
+    std::unique_ptr<v8_inspector::V8Inspector> inspector_;
+    std::unique_ptr<v8_inspector::V8InspectorSession> session_;
+    std::unique_ptr<v8_inspector_channel_wrapper> channel_;
+    v8::Isolate* isolate_;
+    v8::Local<v8::Context> context_;
+    InspectorOnWaitFrontendMessageOnPauseCallback onWaitFrontendMessageOnPause_;
+    bool terminated_;
+    bool run_nested_loop_;
+};
+
+v8_inspector_client_wrapper::v8_inspector_client_wrapper(
+    v8::Platform *platform,
+    const v8::Local<v8::Context> &context,
+    const InspectorOnResponseCallback &onResponse,
+    const InspectorOnWaitFrontendMessageOnPauseCallback &onWaitFrontendMessageOnPause
+) :
+    platform_(platform),
+    context_(context),
+    onWaitFrontendMessageOnPause_(onWaitFrontendMessageOnPause)
+{
+    isolate_ = context->GetIsolate();
+    inspector_ = v8_inspector::V8Inspector::create(isolate_, this);
+    channel_.reset(new v8_inspector_channel_wrapper(isolate_, onResponse));
+    session_ = inspector_->connect(kContextGroupId, channel_.get(), v8_inspector::StringView(), v8_inspector::V8Inspector::kFullyTrusted);
+    context_->SetAlignedPointerInEmbedderData(DEBUGGER_INDEX, this);
+
+    v8_inspector::StringView contextName = convertToStringView("inspector");
+    inspector_->contextCreated(v8_inspector::V8ContextInfo(context_, kContextGroupId, contextName));
+    terminated_ = true;
+    run_nested_loop_ = false;
+}
+
+v8::Isolate* v8_inspector_client_wrapper::getIsolate() {
+    return isolate_;
+}
+
+v8::Local<v8::Context> v8_inspector_client_wrapper::getContext() {
+    return context_;
```

Comment:
I believe we said this is not really needed, if we can avoid it I believe we can avoid saving the context pointer..